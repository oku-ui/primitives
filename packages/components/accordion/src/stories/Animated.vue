<script setup lang="ts">
import { ref, watchEffect } from 'vue'
import { OkuAccordion, OkuAccordionContent, OkuAccordionHeader, OkuAccordionItem, OkuAccordionTrigger } from '@oku-ui/accordion'

const values = ref(['One', 'Two', 'Three', 'Four'])

const count = ref(1)

const hasDynamicContent = ref(false)

const timerRef = ref(0)

watchEffect(() => {
  if (hasDynamicContent.value) {
    timerRef.value = window.setTimeout(() => {
      const nextCount = count.value < 5 ? count.value + 1 : count.value
      if (nextCount === 5)
        hasDynamicContent.value = false
    }, 3000)
    return () => {
      clearTimeout(timerRef.value)
    }
  }
})
function valueChange(event: any) {
  const checked = event.target.checked
  if (checked)
    count.value = 1
  hasDynamicContent.value = checked
}
</script>

<template>
  <div>
    <label>
      <input
        type="checkbox"
        :checked="hasDynamicContent"
        @onChange="valueChange"
      >
      Dynamic content
    </label>
    <h1>Closed by default</h1>
    <OkuAccordion selection-type="single" class="rootClass">
      <OkuAccordionItem v-for="item in values" :key="item" class="itemClass" :value="item">
        <OkuAccordionHeader class="headerClass">
          <OkuAccordionTrigger class="triggerClass">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="animatedContentClass">
          <div v-for="index in count" :key="index" style="padding: 10px">
            Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
            viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
            quam suscipit habitant sed.
          </div>
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
  </div>
</template>
