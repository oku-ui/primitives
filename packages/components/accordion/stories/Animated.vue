<script setup lang="ts">
import { ref, watchEffect } from 'vue'
import { OkuAccordion, OkuAccordionContent, OkuAccordionHeader, OkuAccordionItem, OkuAccordionTrigger } from '@oku-ui/accordion'

const values = ref(['One', 'Two', 'Three', 'Four'])
const count = ref(1)
const hasDynamicContent = ref(false)
const timerRef = ref()

watchEffect(() => {
  if (hasDynamicContent.value) {
    timerRef.value = window.setTimeout(() => {
      const nextCount = count.value < 5 ? count.value + 1 : count.value

      if (nextCount === 5)
        hasDynamicContent.value = false

      count.value = nextCount
    }, 3000)

    return () => {
      clearTimeout(timerRef.value)
    }
  }
})
</script>

<template>
  <div>
    <label>
      <input
        v-model="hasDynamicContent"
        type="checkbox"
      >
      Dynamic content
    </label>
    <h1>Closed by default</h1>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-animated-content">
          <div v-for="index in count" :key="index" style="padding: 10px">
            Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
            viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
            quam suscipit habitant sed.
          </div>
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Open by default</h1>
    <OkuAccordion type="single" class="accordion-root" default-value="One">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-animated-content">
          <div v-for="index in count" :key="index" style="padding: 10px">
            Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
            viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
            quam suscipit habitant sed.
          </div>
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
  </div>
</template>
