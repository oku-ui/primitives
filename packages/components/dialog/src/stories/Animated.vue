<script setup lang="ts">
import { OkuDialog, OkuDialogClose, OkuDialogContent, OkuDialogDescription, OkuDialogOverlay, OkuDialogPortal, OkuDialogTitle, OkuDialogTrigger } from '@oku-ui/dialog'
import { ref } from 'vue'

// TODO: waiting fix focusScope unmounted
const open = ref(false)
</script>

<template>
  <div>
    <OkuDialog v-model="open">
      <OkuDialogTrigger class="dialog-trigger">
        open
      </OkuDialogTrigger>
      <OkuDialogPortal>
        <OkuDialogOverlay class="dialog-animatedOverlay" />
        <OkuDialogContent
          class="dialog-animatedContent"
        >
          <OkuDialogTitle>
            Title
          </OkuDialogTitle>
          <OkuDialogDescription>
            Description
          </OkuDialogDescription>
          <OkuDialogClose>
            close
          </OkuDialogClose>
        </OkuDialogContent>
      </OkuDialogPortal>
    </okudialog>
  </div>
</template>
