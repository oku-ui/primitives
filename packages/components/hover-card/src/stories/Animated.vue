<script setup lang="ts">
import {
  OkuHoverCard,
  OkuHoverCardArrow,
  OkuHoverCardContent,
  OkuHoverCardPortal,
  OkuHoverCardTrigger,
} from '@oku-ui/hover-card'
import CardContentPlaceholder from './CardContentPlaceholder.vue'
</script>

<template>
  <div style="padding: 50px; display: flex; justify-content: center">
    <OkuHoverCard>
      <OkuHoverCardTrigger href="/" class="triggerClass">
        trigger
      </OkuHoverCardTrigger>
      <OkuHoverCardPortal>
        <OkuHoverCardContent class="animatedContentClass" :side-offset="10">
          <OkuHoverCardArrow class="arrowClass" :width="20" :height="10" />
          <CardContentPlaceholder />
        </OkuHoverCardContent>
      </OkuHoverCardPortal>
    </OkuHoverCard>
  </div>
</template>
