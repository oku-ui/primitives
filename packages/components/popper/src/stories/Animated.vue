<script setup lang="ts">
import {
  OkuPopper,
  OkuPopperAnchor,
  OkuPopperArrow,
  OkuPopperContent,
} from '@oku-ui/popper'
import { OkuPortal } from '@oku-ui/portal'
import { ref } from 'vue'
import Scrollable from './Scrollable.vue'

const open = ref(false)
</script>

<template>
  <Scrollable>
    <OkuPopper>
      <OkuPopperAnchor class="anchorClass" @click="open = true">
        open
      </OkuPopperAnchor>

      <OkuPortal v-if="open" as-child>
        <OkuPopperContent class="animatedContentClass" :side-offset="5">
          <button @click="open = false">
            close
          </button>
          <OkuPopperArrow class="arrowClass" :width="20" :height="10" :offset="25" />
        </OkuPopperContent>
      </OkuPortal>
    </OkuPopper>
  </Scrollable>
</template>
