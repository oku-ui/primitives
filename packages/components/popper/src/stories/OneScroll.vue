<script setup lang="ts">
import {
  OkuPopper,
  OkuPopperAnchor,
  OkuPopperArrow,
  OkuPopperContent,
} from '@oku-ui/popper'
import { ref } from 'vue'

const scrollContainer1 = ref<HTMLDivElement | null>(null)
</script>

<template>
  <div
    :style="{
      display: 'flex',
      gap: '100px',
    }"
  >
    <div>
      <h1>In scrolling container</h1>
      <div
        ref="scrollContainer1" :style="{
          width: '400px',
          height: '600px',
          overflow: 'auto',
          border: '1px solid black',
        }"
      >
        <div
          :style="{
            height: '2000px',
          }"
        >
          <OkuPopper>
            <OkuPopperAnchor class="anchorClass small">
              9
            </OkuPopperAnchor>
            <OkuPopperContent
              class="contentClass small" :side-offset="5" hide-when-detached
              :collision-boundary="scrollContainer1"
            >
              <OkuPopperArrow class="arrowClass" :width="10" :height="5" />
              9
            </OkuPopperContent>
          </OkuPopper>
        </div>
      </div>
    </div>
  </div>
</template>
