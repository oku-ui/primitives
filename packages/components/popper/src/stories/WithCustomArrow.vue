<script setup lang="ts">
import {
  OkuPopper,
  OkuPopperAnchor,
  OkuPopperArrow,
  OkuPopperContent,
} from '@oku-ui/popper'
import { ref } from 'vue'
import Scrollable from './Scrollable.vue'
import CustomArrow from './CustomArrow.vue'

const open = ref(false)
</script>

<template>
  <Scrollable>
    <OkuPopper>
      <OkuPopperAnchor class="anchorClass" @click="open = true">
        open
      </OkuPopperAnchor>

      <OkuPopperContent v-if="open" class="contentClass" :side-offset="5" side="right">
        <button @click="open = false">
          close
        </button>
        <OkuPopperArrow as-child :offset="20">
          <CustomArrow :width="20" :height="10" />
        </OkuPopperArrow>
      </OkuPopperContent>
    </OkuPopper>
  </Scrollable>
</template>
