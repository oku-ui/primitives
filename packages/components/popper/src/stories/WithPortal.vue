<script setup lang="ts">
import {
  OkuPopper,
  OkuPopperAnchor,
  OkuPopperArrow,
  OkuPopperContent,
} from '@oku-ui/popper'
import { OkuPortal } from '@oku-ui/portal'
import { ref } from 'vue'
import Scrollable from './Scrollable.vue'

const open = ref(false)
</script>

<template>
  <Scrollable>
    <OkuPopper>
      <OkuPopperAnchor class="popper_anchorClass" @click="open = true">
        open
      </OkuPopperAnchor>

      <OkuPortal v-if="open" as-child>
        <OkuPopperContent class="popper_contentClass" :side-offset="5">
          <button @click="open = false">
            close
          </button>
          <OkuPopperArrow class="popper_arrowClass" :width="20" :height="10" />
        </OkuPopperContent>
      </OkuPortal>
    </OkuPopper>
  </Scrollable>
</template>
