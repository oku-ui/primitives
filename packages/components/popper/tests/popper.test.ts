import { afterEach, beforeEach, describe, expect, it, test, vitest } from 'vitest'
import type { VueWrapper } from '@vue/test-utils'
import { enableAutoUnmount, mount } from '@vue/test-utils'
import { axe } from 'vitest-axe'
import { nextTick } from 'vue'
import {
  OkuPopper,
  OkuPopperAnchor,
  OkuPopperArrow,
  OkuPopperContent,
} from '../src'

import StyledVue from '../src/stories/Styled.vue'
import Animated from '../src/stories/Animated.vue'
import OneScroll from '../src/stories/OneScroll.vue'
import Transition from '../src/stories/Transition.vue'
import WithCustomArrow from '../src/stories/WithCustomArrow.vue'
import WithPortal from '../src/stories/WithPortal.vue'

enableAutoUnmount(afterEach)

describe('OkuPopper', () => {
  it('OkuPopper renders correctly', () => {
    const wrapper = mount(OkuPopper)
    expect(wrapper.html()).toBe('')
  })

  it('OkuPopperArrow renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuPopperArrow)
    expect(() => wrapper()).toThrowError('`OkuPopperArrow` must be used within `OkuPopperContent`')
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuPopperContent)" not found.')
  })

  it('OkuPopperAnchor renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuPopperAnchor)
    expect(() => wrapper()).toThrowError('`OkuPopperAnchor` must be used within `OkuPopper`')
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuPopper)" not found.')
  })

  it('OkuPopperAnchor renders correctly', () => {
    const spy = vitest.spyOn(global.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuPopperContent)
    expect(() => wrapper()).toThrowError('`OkuPopperContent` must be used within `OkuPopper`')
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuPopper)" not found.')
  })

  describe('StyledVue', () => {
    let wrapper: VueWrapper<InstanceType<typeof StyledVue>>

    beforeEach(async () => {
      wrapper = mount(StyledVue, {
        attachTo: document.body,
      })

      const button = wrapper.find('[class="anchorClass"]')
      button.trigger('click')
      await nextTick()
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      const button = wrapper.find('button')
      button.trigger('click')
      await nextTick()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Animated', () => {
    let wrapper: VueWrapper<InstanceType<typeof Animated>>

    beforeEach(async () => {
      wrapper = mount(Animated, {
        props: {
          container: document.body,
        },
        attachTo: document.body,
      })

      const button = wrapper.find('[class="anchorClass"]')
      button.trigger('click')
      await nextTick()
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    test('document.body has correct html', () => {
      expect(document.body.innerHTML).toMatchSnapshot()
    })

    test('should be able to close', async () => {
      const button = document.querySelector('button')
      button?.dispatchEvent(new Event('click'))
      await nextTick()
      expect(document.body.innerHTML).toMatchSnapshot()
    })
  })

  describe('OneScroll', () => {
    let wrapper: VueWrapper<InstanceType<typeof OneScroll>>

    beforeEach(async () => {
      wrapper = mount(OneScroll, {
        attachTo: document.body,
      })
      const button = wrapper.find('[class="anchorClass small"]')
      button.trigger('click')
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    test('renders correctly', () => {
      expect(document.body.innerHTML).toMatchSnapshot()
    })

    test('should be able to close', async () => {
      const button = document.querySelector('button')
      button?.dispatchEvent(new Event('click'))
      await nextTick()
      expect(document.body.innerHTML).toMatchSnapshot()
    })
  })

  describe('Transition', () => {
    let wrapper: VueWrapper<InstanceType<typeof Transition>>

    beforeEach(async () => {
      wrapper = mount(Transition, {
        attachTo: document.body,
      })
      const button = wrapper.find('[class="anchorClass"]')
      button.trigger('click')
      await new Promise(resolve => setTimeout(resolve, 300))
      await nextTick()
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    test('renders correctly', () => {
      expect(document.body.innerHTML).toMatchSnapshot()
    })

    test('should be able to close', async () => {
      const button = document.querySelector('button')
      button?.click()
      await nextTick()
      // TODO: fix this dont see close button
      expect(document.body.innerHTML).toMatchSnapshot()
    })
  })

  describe('WithCustomArrow', () => {
    let wrapper: VueWrapper<InstanceType<typeof WithCustomArrow>>

    beforeEach(async () => {
      wrapper = mount(WithCustomArrow, {
        attachTo: document.body,
      })
      const button = wrapper.find('[class="anchorClass"]')
      button.trigger('click')
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('WithPortal', () => {
    let wrapper: VueWrapper<InstanceType<typeof WithPortal>>

    beforeEach(async () => {
      wrapper = mount(WithPortal, {
        attachTo: document.body,
      })

      const button = wrapper.find('[class="anchorClass"]')
      button.trigger('click')
    })

    /**
    * @vitest-environment jsdom
    */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
