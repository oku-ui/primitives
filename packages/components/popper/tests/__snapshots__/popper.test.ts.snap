// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OkuPopper > Animated > document.body has correct html 1`] = `"<div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"><div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\" container=\\"[object HTMLBodyElement]\\"><div class=\\"popper_anchorClass\\"> open </div><!--teleport start--><!--teleport end--></div></div><div data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;\\"><div data-side=\\"bottom\\" data-align=\\"center\\" class=\\"popper_animatedContentClass\\" style=\\"animation: none;\\"><button> close </button><span style=\\"position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;\\"><svg class=\\"popper_arrowClass\\" offset=\\"25\\" style=\\"display: block;\\" width=\\"20\\" height=\\"10\\" viewBox=\\"0 0 30 10\\" preserveAspectRatio=\\"none\\"><polygon points=\\"0,0 30,0 15,10\\"></polygon></svg></span></div></div>"`;

exports[`OkuPopper > Animated > renders correctly 1`] = `
"<div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\" container=\\"[object HTMLBodyElement]\\">
  <div class=\\"popper_anchorClass\\"> open </div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`OkuPopper > Animated > should be able to close 1`] = `"<div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"><div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\" container=\\"[object HTMLBodyElement]\\"><div class=\\"popper_anchorClass\\"> open </div><!--v-if--></div></div>"`;

exports[`OkuPopper > OneScroll > renders correctly 1`] = `"<div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"><div style=\\"display: flex; gap: 100px;\\"><div><h1>In scrolling container</h1><div style=\\"width: 400px; height: 600px; overflow: auto; border: 1px solid black;\\"><div style=\\"height: 2000px;\\"><div class=\\"popper_anchorClass small\\"> 9 </div><div data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;\\"><div data-side=\\"bottom\\" data-align=\\"center\\" class=\\"popper_contentClass small\\" style=\\"animation: none;\\"><span style=\\"position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;\\"><svg class=\\"popper_arrowClass\\" style=\\"display: block;\\" width=\\"10\\" height=\\"5\\" viewBox=\\"0 0 30 10\\" preserveAspectRatio=\\"none\\"><polygon points=\\"0,0 30,0 15,10\\"></polygon></svg></span> 9 </div></div></div></div></div></div></div>"`;

exports[`OkuPopper > OneScroll > should be able to close 1`] = `"<div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"><div style=\\"display: flex; gap: 100px;\\"><div><h1>In scrolling container</h1><div style=\\"width: 400px; height: 600px; overflow: auto; border: 1px solid black;\\"><div style=\\"height: 2000px;\\"><div class=\\"popper_anchorClass small\\"> 9 </div><div data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;\\"><div data-side=\\"bottom\\" data-align=\\"center\\" class=\\"popper_contentClass small\\" style=\\"animation: none;\\"><span style=\\"position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;\\"><svg class=\\"popper_arrowClass\\" style=\\"display: block;\\" width=\\"10\\" height=\\"5\\" viewBox=\\"0 0 30 10\\" preserveAspectRatio=\\"none\\"><polygon points=\\"0,0 30,0 15,10\\"></polygon></svg></span> 9 </div></div></div></div></div></div></div>"`;

exports[`OkuPopper > StyledVue > renders correctly 1`] = `
"<div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\">
  <div class=\\"popper_anchorClass\\"> open </div>
  <div data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;\\">
    <div data-side=\\"bottom\\" data-align=\\"center\\" class=\\"popper_contentClass\\" style=\\"animation: none;\\"><button> close </button><span style=\\"position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;\\"><svg class=\\"popper_arrowClass\\" style=\\"display: block;\\" width=\\"20\\" height=\\"10\\" viewBox=\\"0 0 30 10\\" preserveAspectRatio=\\"none\\"><polygon points=\\"0,0 30,0 15,10\\"></polygon></svg></span></div>
  </div>
</div>"
`;

exports[`OkuPopper > StyledVue > should be able to close 1`] = `
"<div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\">
  <div class=\\"popper_anchorClass\\"> open </div>
  <!--v-if-->
</div>"
`;

exports[`OkuPopper > Transition > renders correctly 1`] = `
"<div data-v-21f44795=\\"\\" class=\\"popper_anchorClass\\"> Open </div>
<!--teleport start-->
<!--teleport end-->"
`;

exports[`OkuPopper > Transition > renders correctly 2`] = `"<div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"><div data-v-21f44795=\\"\\" class=\\"popper_anchorClass\\"> Open </div><!--teleport start--><!--teleport end--></div><transition-stub data-v-21f44795=\\"\\" name=\\"fade\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\"><div data-v-21f44795=\\"\\" data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(5px, 0px); min-width: max-content; --oku-popper-transform-origin: 0px 0px; --oku-popper-available-width: -5px; --oku-popper-available-height: 0px; --oku-popper-anchor-width: 0px; --oku-popper-anchor-height: 0px;\\"><div data-side=\\"right\\" data-align=\\"start\\" class=\\"popper_animatedContentClass\\" style=\\"\\"><button data-v-21f44795=\\"\\"> close </button><span data-v-21f44795=\\"\\" style=\\"position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%);\\"><svg class=\\"popper_arrowClass\\" offset=\\"25\\" style=\\"display: block;\\" width=\\"20\\" height=\\"10\\" viewBox=\\"0 0 30 10\\" preserveAspectRatio=\\"none\\"><polygon points=\\"0,0 30,0 15,10\\"></polygon></svg></span></div></div></transition-stub>"`;

exports[`OkuPopper > Transition > should be able to close 1`] = `"<div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"></div><div data-v-app=\\"\\"><div data-v-21f44795=\\"\\" class=\\"popper_anchorClass\\"> Open </div><!--teleport start--><!--teleport end--></div><transition-stub data-v-21f44795=\\"\\" name=\\"fade\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\"><div data-v-21f44795=\\"\\" data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(5px, 0px); min-width: max-content; --oku-popper-transform-origin: 0px 0px; --oku-popper-available-width: -5px; --oku-popper-available-height: 0px; --oku-popper-anchor-width: 0px; --oku-popper-anchor-height: 0px; display: none;\\"><div data-side=\\"right\\" data-align=\\"start\\" class=\\"popper_animatedContentClass\\" style=\\"\\"><button data-v-21f44795=\\"\\"> close </button><span data-v-21f44795=\\"\\" style=\\"position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%);\\"><svg class=\\"popper_arrowClass\\" offset=\\"25\\" style=\\"display: block;\\" width=\\"20\\" height=\\"10\\" viewBox=\\"0 0 30 10\\" preserveAspectRatio=\\"none\\"><polygon points=\\"0,0 30,0 15,10\\"></polygon></svg></span></div></div></transition-stub>"`;

exports[`OkuPopper > WithCustomArrow > renders correctly 1`] = `
"<div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\">
  <div class=\\"popper_anchorClass\\"> open </div>
  <div data-oku-popper-content-wrapper=\\"\\" style=\\"position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;\\">
    <div data-side=\\"right\\" data-align=\\"center\\" class=\\"popper_contentClass\\" style=\\"animation: none;\\"><button> close </button><span style=\\"position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%); visibility: hidden;\\"><div style=\\"display: block; width: 20px; height: 10px; border-bottom-left-radius: 10px; border-bottom-right-radius: 10px; background-color: tomato;\\" width=\\"10\\" height=\\"5\\" offset=\\"20\\" viewbox=\\"0 0 30 10\\" preserveaspectratio=\\"none\\"></div></span></div>
  </div>
</div>"
`;

exports[`OkuPopper > WithPortal > renders correctly 1`] = `
"<div style=\\"display: flex; align-items: center; justify-content: center; height: 200vh;\\">
  <div class=\\"popper_anchorClass\\"> open </div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
